### DISCUSSION TITLE
Download XCode via dev portal

### DISCUSSION THREAD

**Discussion by blacktop** [fastlane/fastlane#19081] [Q&A]
Would it be possible (are their API endpoints) to download beta XCode via fastlane?  Do you know if there exists or where I would look to find those API endpoints to talk to the Downloads part of the dev portal?

Thank you!!

**Comment by blacktop** [comment]
I've discovered `https://developer.apple.com/services-account/QH65B2/downloadws/listDownloads.action` but that is just for the **All** section and is things like KDKs and command line tools etc

**Comment by lucgrabowski** [comment]
There is an action to install xcode: http://docs.fastlane.tools/actions/xcode_install/#xcode_install

**Reply by blacktop** [reply]
I believe that is just for the Release XCodes.  Is there a way to download the BETA Xcodes? Would be nice to grab the beta ipsws too via an official Apple API endpoint?

**Reply by lucgrabowski** [reply]
It would require checking. I've briefly checked `xcversion list` as written on github https://github.com/xcpretty/xcode-install (used internally be the `xcode_install` action):
And I got a list of Xcodes:
```
...
12.3
12.4 Release Candidate
12.4
12.5 beta
12.5 beta 2
12.5 beta 3
12.5 Release Candidate
12.5
12.5.1
13 beta
13 beta 2
```
So it looks promising
